lock/src/main/java/com/auth0/android/lock/provider/ProviderResolver.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * IdentityProviderHandler.java
+ *
+ * Copyright (c) 2016 Auth0 (http://auth0.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.auth0.android.lock.provider;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+
+public interface ProviderResolver {
+
+    /**
+     * Requests a custom IdentityProvider to use with the given connection.
+     *
+     * @param context        a valid context.
+     * @param callback       the callback Lock will call with the authentication result.
+     * @param connectionName the requested connection name
+     * @return the identity provider to use with this connection, or null to use the default.
```

Comment:
`or null to use the default` to `or null if you there is no associated provider for the connection name`